benchmarks/vector_db_compare/test_system_qualification.py

```
@@ -0,0 +1,253 @@
+            "reason": "official deployment unavailable",
+            "boundary": self.plan["rows"][3]["boundary"],
+            "identity": result["rows"][3]["identity"],
+        }
+        self.assertFalse(validate_result(self.plan, self.plan_sha256, result))
+        with self.assertRaisesRegex(ContractError, "every row"):
+            validate_result(self.plan, self.plan_sha256, result, require_complete=True)
+
+    def test_search_matrix_and_metrics_fail_closed(self) -> None:
+        missing_cell = self.result()
+        missing_cell["rows"][4]["corpora"][0]["repetitions"][0]["searches"].pop()
+        with self.assertRaisesRegex(ContractError, "search matrix"):
```

Ruff (0.16.1)

[warning] Use `pytest.raises` instead of unittest-style `assertRaisesRegex`

Replace `assertRaisesRegex` with `pytest.raises`

(PT027)